src/Framework/App.Runtime/src/Microsoft.AspNetCore.App.Runtime.csproj

```diff
@@ -478,7 +478,7 @@ This package is an internal implementation of the .NET Core SDK and is not meant
   </Target>
 
   <Target Name="_GenerateComposites"
-          Condition="'$(BuildOsName)' == 'linux'"
+          Condition="'$(TargetOsName)' == 'linux' or '$(TargetOsName)' == 'linux-musl'"
```

Comment:
so assume musl doesnt get built on vanilla linux?

Reply:
I think it's the other way around - we need to cater for the fact that the targeting OS may slightly differ from the build OS so that e.g. we end up building images targeting linux-musl on vanilla linux.

Reply:
Can you please elaborate how the change around line 106 is supposed to change matters? Frankly speaking, to me it looks like just a formatting change adding extra blanks that shouldn't affect actual msbuild condition evaluation.

Reply:
Oh it is just a formatting change here. What I meant is that because we use an Alpine musl container to work in that case, `BuildOsName` is special cased to become `linux` for any non-x64 `linux-musl` case. The comment above (in the code, not in this PR conversation) explains it in more detail.

Reply:
I see, thanks for clarifying, it makes more sense to me now!

Reply:
Would a `.Contains` work here, or are there other potential linux `TargetOsName`s we want to avoid?

Reply:
Do we have plans of supporting any other linux flavor? @trylek @mangod9

Reply:
lets keep it limited to these two for which we produce container images. We can evaluate on how the rollout of these go in early previews.